cros_gralloc/cros_gralloc_driver.cc

```diff
@@ -109,6 +114,96 @@ cros_gralloc_driver *cros_gralloc_driver::get_instance()
         close(fd);                   \
     }
 
+int32_t cros_gralloc_driver::reload()
+{
+	int fd;
+	drmVersionPtr version;
+	char const *str = "%s/renderD%d";
+	uint32_t min_node = 128;
+	uint32_t max_node = (min_node + drv_num_ + IVSH_DEVICE_NUM);
+	uint32_t j;
+	char *node;
+
+	const int render_num = 10;
+	const int name_length = 50;
+	int availabe_node = 0;
+	uint32_t gpu_grp_type = 0;
+
+	availabe_node = 0;
+	for (uint32_t i = min_node + drv_num_; i < max_node; i++) {
+		if (asprintf(&node, str, DRM_DIR_NAME, i) < 0)
+			continue;
+
+		fd = open(node, O_RDWR, 0);
+		free(node);
+		if (fd < 0)
+			continue;
+
+		version = drmGetVersion(fd);
+		if (!version) {
+			close(fd);
+			continue;
+		}
+
+		availabe_node++;
+		close(fd);
+		drmFreeVersion(version);
+	}
+
+	// have the ivshemem node
+	if ((availabe_node > 0) && !drv_ivshmem_) {
+		drv_num_ = drv_num_ + availabe_node;
```

Comment:
If I understand correctly, you are searching for the render node with largest id, but it seems the result should't be calculated as this. If there is an empty slot, the result is incorrect. For example, if we have renderD128, renderD129, and renderD131 (somehow the renderD130 node is not present), you will try to open renderD130.

Reply:
i don't understand in which case there is empty slot.

Reply:
change accordingly, please check.